arch/arm64/boot/dts/qcom/sdm450-samsung-a6plte-r4.dts

```diff
@@ -0,0 +1,532 @@
+	};
+
+	prox_led_vdd: vreg-prox-led-3v3 {
+		compatible = "regulator-fixed";
+		regulator-name = "proximity_led_vdd";
+		regulator-min-microvolt = <3300000>;
+		regulator-max-microvolt = <3300000>;
+		startup-delay-us = <1000>;
+
+		gpio = <&tlmm 127 GPIO_ACTIVE_HIGH>;
+
+		enable-active-high;
+	};
+
+	grip_vdd: vreg-grip-sensor-2v8 {
+		compatible = "regulator-fixed";
+		regulator-name = "grip_sensor_vdd";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		startup-delay-us = <1000>;
+
+		gpio = <&tlmm 87 GPIO_ACTIVE_HIGH>;
+
+		enable-active-high;
+	};
+
+	vddio_cam: vreg-cam-io-1v8 {
+		compatible = "regulator-fixed";
+		regulator-name = "cam_vddio";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+		startup-delay-us = <1000>;
+
+		gpio = <&expander 12 GPIO_ACTIVE_HIGH>;
+
+		vin-supply = <&vdd_cam_aux>;
+
+		enable-active-high;
+	};
+
+	avdd_cam_front: vreg-cam-avdd-2v8 {
+		compatible = "regulator-fixed";
+		regulator-name = "cam_avdd_front";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		startup-delay-us = <1000>;
+
+		gpio = <&expander 13 GPIO_ACTIVE_HIGH>;
+
+		enable-active-high;
+	};
+
+	vdd_cam_main: vreg-cam-main-vdd-2p8 {
+		compatible = "regulator-fixed";
+		regulator-name = "cam_vdd_main";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		startup-delay-us = <1000>;
+
+		gpio = <&tlmm 92 GPIO_ACTIVE_HIGH>;
+
+		enable-active-high;
+	};
+
+	vdd_cam_aux: vreg-cam-aux-io-2p8 {
+		compatible = "regulator-fixed";
+		regulator-name = "cam_aux_vddio";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		startup-delay-us = <1000>;
+
+		gpio = <&tlmm 94 GPIO_ACTIVE_HIGH>;
+
+		enable-active-high;
+	};
+
+	vdd_actuator: vreg-cam-lens-2p8 {
+		compatible = "regulator-fixed";
+		regulator-name = "cam_lens_vdd";
+		regulator-min-microvolt = <2800000>;
+		regulator-max-microvolt = <2800000>;
+		startup-delay-us = <1000>;
+
+		gpio = <&tlmm 89 GPIO_ACTIVE_HIGH>;
+
+		enable-active-high;
+	};
+};
+
+&hsusb_phy {
+	vdd-supply = <&pm8953_l3>;
+	vdda-pll-supply = <&pm8953_l7>;
+	vdda-phy-dpdm-supply = <&pm8953_l13>;
+
+	status = "okay";
+};
+
+&i2c_2 {
+	status = "okay";
+
+	touchscreen@48 {
+		compatible = "samsung,s6sy761";
+		reg = <0x48>;
+
+		interrupt-parent = <&tlmm>;
+		interrupts = <59 IRQ_TYPE_EDGE_FALLING>;
+
+		avdd-supply = <&ts_avdd>;
+		vdd-supply = <&ts_vdd>;
+
+		touchscreen-size-x = <4096>;
+		touchscreen-size-y = <4096>;
+	};
+};
+
+&i2c_3 {
+	status = "okay";
+
+	front_calib_eeprom: eeprom@50 {
+		compatible = "atmel,24c64";
+		reg = <0x50>;
+
+		label = "front-camera-eeprom";
+		vcc-supply = <&vdd_cam_main>;
+		// vio-supply = <&vddio_cam>; // FIXME unused
+	};
+
+	main_calib_eeprom: eeprom@51 {
+		compatible = "atmel,24c128";
+		reg = <0x51>;
+
+		label = "main-camera-eeprom";
+		vcc-supply = <&vdd_cam_main>;
+		// vio-supply = <&vddio_cam>; // FIXME unused
```

Comment:
@vldly  (and above)

Reply:
Wasn't going to upstream it because those EEPROMs have proprietary format anyway. Also this binding allows write but it should be read only.